### PR TITLE
Install dependencies and report errors during connection test (#903)

### LINKED ISSUE #903: SQLTools not asking to install the dependencies and not validating form with the button "Test Connection"
Based on this [discussion](https://github.com/mtxr/vscode-sqltools/discussions/785) basically SQLTools only asks to install extra dependencies when you click on the button "Save Connection" and then the button "Connect now", meaning that the button "Test Connection" is always gonna fail if you don't have the extra dependencies already installed or do the steps mentioned first.

Also, the button "Test Connection" doesn't do any validation on the form, all fields could be empty and u can still "test" the connection.

### PR DESCRIPTION
**Pull request by gjsjohnmurray**
This PR fixes the problems reported in #903.

----

Thank you for your contribution! 
Before submitting this PR, please make sure:

- [X] Your code builds clean without any errors or warnings
- [X] You have made the needed changes to the docs
- [X] You have written a description of what is the purpose of this pull request above

### CODE REVIEW

File: packages/plugins/connection-manager/language-server.ts [hunk 1]
```diff
@@ -197,7 +198,12 @@ export default class ConnectionManagerPlugin implements ILanguageServerPlugin {
       progressBase && this.server.sendNotification(ProgressNotificationComplete, progressBase);
       e = decorateLSException(e, { conn: req.conn });
       if (e.data && e.data.notification) {
-        delete e.data.args.conn;
+
+        // To fix https://github.com/mtxr/vscode-sqltools/issues/903 we need not to delete conn from the notification.
+        // Unclear in what circumstances we *do* need to delete it, but made the minimal change to fix #903.
+        if (e.data.notification !== MissingModuleNotification) {
+          delete e.data.args.conn;
```
Comment (mtxr) [inline packages/plugins/connection-manager/language-server.ts]: @gjsjohnmurray the reason to delete credentials here is to avoid any kind of error tracking tool to log passwords. In the very beginning of the extension, I recall to be testing and I realized the extensions share context and a I believe it still remains like this.

That said, one extension (for python snippets) had a global error handling that was unhandled errors thrown by any extension, to avoid any mistakes, I decided to delete the connection information from the notification here.

It's just that. That reminded me how dangerous it can be to monitor errors even if I want to track errors/bugs on sqltools

Ever since, even for SQLTools I stopped to send events to sentry and telemetry from sqltools is fully disabled, see
CleanShot 2022-09-11 at 02 44 55

I'll remove that code completely since it's just weight we are carrying, but those are the reasons behind this weird delete...

File: packages/plugins/connection-manager/language-server.ts [hunk 2]
```diff
@@ -197,7 +198,12 @@ export default class ConnectionManagerPlugin implements ILanguageServerPlugin {
...
+        }
```
Comment (mtxr) [inline packages/plugins/connection-manager/language-server.ts]:
```suggestion
```

**Comment by mtxr** [comment]
@gjsjohnmurray everything lgtm, but before merging, check my previous message, I believe we can just get rid of that weird piece of code. let me know your thoughts

File: packages/plugins/connection-manager/language-server.ts (hunk 2 above)
Comment (gjsjohnmurray) [reply packages/plugins/connection-manager/language-server.ts]: Thanks for the explanation. I just pushed an equivalent change which also removes the redundant comment.

**Review by mtxr — APPROVED** [review]
